src/Microsoft.Extensions.Configuration/ConfigurationRoot.cs

```diff
@@ -24,6 +24,12 @@ public ConfigurationRoot(IList<IConfigurationProvider> providers)
             _providers = providers;
         }
 
+        public string Key => string.Empty;
+
+        public string Path => string.Empty;
+
+        public string Value { get; set; }
```

Comment:
But my counter argument would be, what is the point of diminishing returns? It is starting to look more and more like we are trying to shove these additional members on the root that really don't make any sense. In other words, there is a reason these member don't belong in the root that are stronger than just hiding them from intellisense. Yet another way to put it: that ideal place in which you can treat all IConfiguration instances uniformly just isn't real :smile: